src/hw_isolation_event/hw_status_manager.cpp

```diff
@@ -460,6 +460,55 @@ void Manager::clearHwStatusEventIfexists(const std::string& hwInventoryPath)
     });
 }
 
+void Manager::handleDeallocatedHw()
+{
+    auto deallocatedHw = std::move(_deallocatedHwHandler.front());
+    _deallocatedHwHandler.pop();
+    if (deallocatedHw.second->isEnabled())
+    {
+        deallocatedHw.second->setEnabled(false);
+    }
```

Comment:
do we need this as the timer will be automatically removed?

Reply:
if it takes more than 5 sec to exit from the local scope then, the timer will call again but, we don't need to do that since we are already in the mid of the process.
FYR: https://www.freedesktop.org/software/systemd/man/sd_event_source_set_enabled.html

By default, [sdeventplus setting is enabled](https://github.com/openbmc/sdeventplus/blob/master/src/sdeventplus/utility/timer.cpp#L31) since we are passing the interval while creating the timer object.